app/controllers/shopping_lists_controller.rb

```diff
@@ -0,0 +1,23 @@
+class ShoppingListsController < ApplicationController
+  def index
+    @missing_foods = current_user.foods.where.not(id: RecipeFood.includes(:recipe)
+    .where(recipes: { user_id: current_user.id }).pluck(:food_id))
```

Comment:
- [ Optional]  like you fixed N+1 queries problem here, I recommend you to do that in other queries. This will protect your application to provide un necessary queries